src/util/dbid.cpp

```diff
@@ -0,0 +1,16 @@
+#include "util/dbid.h"
+
+
+const QVariant::Type DbId::kVariantType = DbId().toVariant().type();
+
+//static
+DbId::value_type DbId::valueOf(QVariant /*pass-by-value*/ variant) {
+    if (variant.canConvert<value_type>() &&
+            variant.convert(kVariantType)) {
+        value_type value = variant.value<value_type>();
```

Comment:
The extra copy of QVariant should not hurt (Qt uses implicit sharing if the 
compiler is not able to optimize) and the generic approach with convert() is 
much more elegant and maintainable.

On 08/24/2015 03:02 PM, Daniel Schürmann wrote:
>
> In src/util/dbid.cpp 
> <https://github.com/mixxxdj/mixxx/pull/684#discussion_r37748022>:
>
> > @@ -0,0 +1,16 @@
> > +#include "util/dbid.h"
> > +
> > +
> > +const QVariant::Type DbId::kVariantType = DbId().toVariant().type();
> > +
> > +//static
> > +DbId::value_type DbId::valueOf(QVariant /*pass-by-value*/ variant) {
> > +    if (variant.canConvert<value_type>() &&
> > +            variant.convert(kVariantType)) {
> > +        value_type value = variant.value<value_type>();
>
> How about just using "int toInt(bool *ok = 0) const"
>
> IMHO here is performance for beeing able to change the type in future 
> (which will be unlikely to happen).
> This class is used very often thougout the Mixxx source.
>
> If you like to keep the flexibility, we may turn DbId into a template and 
> implement DbId::valueOf() by toInt()
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/mixxxdj/mixxx/pull/684/files#r37748022>.
>